### PR TITLE
Move test beatmap from TestCasePlayer into an instantiable class

### PR DESCRIPTION
**Pull request by smoogipoo**
So that we can use this beatmap elsewhere (editor).

### CODE REVIEW

File: osu.Game/Tests/Beatmaps/TestBeatmap.cs [hunk 1]
```diff
@@ -0,0 +1,725 @@
...
+223420,-100,4,2,0,30,0,0
+224357,-100,4,2,0,25,0,0
+225295,-100,4,2,0,20,0,0
+226232,-100,4,2,0,15,0,0
+226701,-100,4,2,0,10,0,0
+227170,-100,4,2,0,5,0,0
+
+
+[Colours]
+        Combo1 : 17,254,176
```
Comment (peppy) [inline osu.Game/Tests/Beatmaps/TestBeatmap.cs]: what happen here o_O

File: osu.Game/Tests/Beatmaps/TestWorkingBeatmap.cs [hunk 2]
```diff
@@ -1,24 +1,46 @@
...
+
+        protected override Track GetTrack()
+        {
+            var lastObject = beatmap.HitObjects.LastOrDefault();
+            if (lastObject != null)
+                return new TestTrack(((lastObject as IHasEndTime)?.EndTime ?? lastObject.StartTime) + 1000);
+            return new TrackVirtual();
+        }
+
+        private class TestTrack : TrackVirtual
```
Comment (peppy) [inline osu.Game/Tests/Beatmaps/TestWorkingBeatmap.cs]: planning on adding more to this class? if not, could possibly just set the length directly

Comment (smoogipoo) [reply osu.Game/Tests/Beatmaps/TestWorkingBeatmap.cs]: Length setter is protected (probably for a good reason). That's the only reason this class exists.

Comment (peppy) [reply osu.Game/Tests/Beatmaps/TestWorkingBeatmap.cs]: hmm, interesting. i get the feeling we'll need to unprotect at some point, but this is fine for now i guess!

**Review by peppy — APPROVED** [review]

**Pull request merged**